types/jest/index.d.ts

```diff
@@ -891,12 +949,18 @@ declare namespace jest {
         toReturnTimes(count: number): R;
         /**
          * Ensure that a mock function has returned a specified value at least once.
+         *
+         * Optionally, you can provide a type for the expected value via a generic.
+         * This is particuarly useful for ensuring expected objects have the right structure.
          */
-        toReturnWith(value: any): R;
+        toReturnWith<E>(value: E): R;
```

Comment:
```suggestion
        toReturnWith<E = any>(value: E): R;
```

Explicit fallback rather than implicit